File: kata-trivia-java/src/main/java/trivia/Player.java
package trivia;

class Player {
   final String name;
   int position = 1;
   int coins = 0;
   boolean inPenaltyBox = false;

   Player(String name) {
      this.name = name;
   }

   boolean hasName(String otherName) {
      return name.equals(otherName);
   }

   void move(int roll, int boardSize) {
      position = position + roll;
      if (position > boardSize) position = position - boardSize;
   }

   void addCoin() {
      coins++;
   }

   void sendToPenaltyBox() {
      inPenaltyBox = true;
   }

   void releaseFromPenaltyBox() {
      inPenaltyBox = false;
   }

   boolean hasWon(int winningCoins) {
      return coins == winningCoins;
   }
}

File: kata-trivia-python/trivia/player.py
class Player:
    def __init__(self, name):
        self.name = name
        self.position = 1
        self.coins = 0
        self.in_penalty_box = False

    def has_name(self, other_name):
        return self.name == other_name

    def move(self, roll, board_size):
        self.position += roll
        if self.position > board_size:
            self.position -= board_size

    def add_coin(self):
        self.coins += 1

    def send_to_penalty_box(self):
        self.in_penalty_box = True

    def release_from_penalty_box(self):
        self.in_penalty_box = False

    def has_won(self, winning_coins):
        return self.coins == winning_coins
